src/hooks/useTxSender.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react'
+import * as txSender from '@/services/tx/tx-sender'
+import { useSafeSDK } from '@/hooks/coreSDK/safeCoreSDK'
+
+// Extends the return type of each txSender function by undefined
+type NullableTxSenderFunctions = {
+  [K in keyof typeof txSender]: (...args: Parameters<typeof txSender[K]>) => undefined | ReturnType<typeof txSender[K]>
+}
+
+const useTxSender = (): NullableTxSenderFunctions => {
+  const sdk = useSafeSDK()
+
+  // @ts-ignore
+  return useMemo(
+    () =>
+      Object.keys(txSender).reduce((result, key) => {
+        // @ts-ignore
```

Comment:
I was able to remove one ts-ignore but the other one proves difficult because `txSender` functions expect various number of arguments so it doesn't allow just sprading `...args`. Any ideas?

Reply:
Perhaps this may work?

```ts
result[key] = sdk ? txSender[key] : undefined
```

Reply:
Had to adjust it so its a function that returns undefined but this way it works, thanks!